ArduCopter/mode_circle.cpp

```diff
@@ -39,6 +39,56 @@ void ModeCircle::run()
         pilot_yaw_override = true;
     }
 
+    // pilot changes to circle rate and radius
+    // skip if in radio failsafe
+    if (!copter.failsafe.radio && !is_zero(copter.circle_nav->get_control())) {
+        // update the circle controller's radius target based on pilot pitch stick inputs
+        float radius_current = copter.circle_nav->get_radius();         // circle controller's radius target, which begins as the circle_radius parameter
+        float pitch_stick = channel_pitch->norm_input_dz();             // pitch stick normalized -1 to 1
+        float nav_speed = copter.wp_nav->get_default_speed_xy();        // copter loiter speed
```

Comment:
re the comment, this probably isn't really the "loiter speed".  It's the horizontal speed during waypoint navigation... or the WPNAV_SPEED parameter value.

Reply:
Done